boa/src/builtins/array/mod.rs

```diff
@@ -1334,6 +1335,195 @@ impl Array {
+    /// An Array object containing the deleted elements (if any) is returned.
+
+    pub(crate) fn splice(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        // 1. Let O be ? ToObject(this value).
+        let o = this.to_object(context)?;
+        // 2. Let len be ? LengthOfArrayLike(O).
+        let len = this.get_field("length", context)?.to_length(context)?;
+        // 3. Let relativeStart be ? ToIntegerOrInfinity(start).
+        // 4. If relativeStart is -∞, let actualStart be 0.
+        // 5. Else if relativeStart < 0, let actualStart be max(len + relativeStart, 0).
+        // 6. Else, let actualStart be min(relativeStart, len).
+        let actual_start = Self::get_relative_start(context, args.get(0), len)?;
+        // 7. If start is not present, then
+        let insert_count = if args.get(0).is_none() || args.get(1).is_none() {
+            // 7a. Let insertCount be 0.
+            // 8a. Let insertCount be 0.
+            0
+        } else {
+            // 9a. Let insertCount be the number of elements in items.
+            args.len()
+        };
+        let actual_delete_count = if args.get(0).is_none() {
+            // 7b. Let actualDeleteCount be 0.
+            0
+        } else if args.get(1).is_none() {
+            // 8b. Let actualDeleteCount be len - actualStart.
+            len - actual_start
+        } else {
+            // 9b. Let dc be ? ToIntegerOrInfinity(deleteCount).
+            let dc = args.get(1).ok_or(0)?.to_integer_or_infinity(context)?;
+            // c. Let actualDeleteCount be the result of clamping dc between 0 and len - actualStart.
+            let max = len - actual_start;
+            match dc {
+                IntegerOrInfinity::Integer(i) => {
+                    if i < 0 {
+                        0
+                    } else if i as usize > max {
+                        max
+                    } else {
+                        i as usize
+                    }
+                }
+                IntegerOrInfinity::PositiveInfinity => max,
+                IntegerOrInfinity::NegativeInfinity => 0,
+            }
+        };
+        // 10. If len + insertCount - actualDeleteCount > 253 - 1, throw a TypeError exception.
+        if len + insert_count - actual_delete_count > Number::MAX_SAFE_INTEGER as usize {
+            return context.throw_type_error("Target splice exceeded max safe integer value");
+        }
+        // 11. Let A be ? ArraySpeciesCreate(O, actualDeleteCount).
+        let arr = Self::array_species_create(&o, actual_delete_count as u32, context)?;
+        // 12. Let k be 0.
+        // 13. Repeat, while k < actualDeleteCount,
+        for k in 0..actual_delete_count {
+            // a. Let from be ! ToString(𝔽(actualStart + k)).
+            // b. Let fromPresent be ? HasProperty(O, from).
+            let from_present = this.has_field(actual_start + k);
```

Comment:
```suggestion
            // b. Let fromPresent be ? HasProperty(O, from).
            let from_present = o.has_property(actual_start + k, context)?;
```
call on `o` and use `has_proeprty`